hw/pci-host/pci_lite.c

```diff
@@ -230,15 +225,23 @@ PCIBus *pci_lite_init(MemoryRegion *address_space_mem,
     //object_property_add_child(qdev_get_machine(), "pcilite", OBJECT(dev), NULL);
     qdev_init_nofail(dev);
 
-    range_set_bounds(&pci_lite->pci_hole,
-                    PCI_LITE_PCIEXBAR_BASE + PCI_LITE_PCIEXBAR_SIZE,
-                    IO_APIC_DEFAULT_ADDRESS);
+    uint64_t pci_hole_base, pci_hole_end, mcfg_base;
+
+    mcfg_base = PCI_HOST_PCIEXBAR_BASE + nr * PCI_HOST_PCIEXBAR_SIZE;
+    pci_hole_base = PCI_HOST_PCI_HOLE_BASE + nr * PCI_HOST_PCI_HOLE_SIZE;
+    pci_hole_end = pci_hole_base + PCI_HOST_PCI_HOLE_SIZE;
+    /* TODO: Do we still need IOAPIC and LAPCI spaces since using kernel_irqchip? */
```

Comment:
@liujing2 yes. we still need to keep these addresses as they are reserved. The 8MB range above is also mapped for MSI.

Reply:
OK, got it. KVM uses the address, I see.